test/service/test_spreadsheet_generator.py

```diff
@@ -195,39 +202,53 @@ def test_generate(self):
+                               "Donor organism", "Collection protocol", "Specimen from organism", "Organoid",
+                               "Cell line",
                                "Imaged specimen", "Dissociation protocol", "Aggregate generation protocol",
                                "Differentiation protocol", "Ipsc induction protocol", "Cell suspension",
-                               "Imaging protocol", "Imaging protocol - Channel","Imaging protocol - Probe",
+                               "Imaging protocol", "Imaging protocol - Channel", "Imaging protocol - Probe",
                                "Imaging preparation protocol", "Image file", "Library preparation protocol",
                                "Sequencing protocol", "Supplementary file", "Sequence file", "Schemas"]
 
         self.assertEqual(actual_tab_names, expected_tab_names1)
 
         # check the row headers are present and correct.
         # note: it is difficult to extend this test to multiple tabs (too many expected row headers to list here). Any ideas/thoughts about how to extend it welcome.
-        xls = pd.ExcelFile("ss2.xlsx")
-        df = pd.read_excel(xls, "Project")
+
+        df = pd.read_excel(self.test_file, sheet_name='Project', engine='openpyxl')
         self.assertEqual(df.columns[0], "PROJECT LABEL (Required)")
-        self.assertEqual(df.iloc[0,0], "A short name for the project.")
-        self.assertEqual(df.iloc[2,0], "project.project_core.project_short_name")
+        self.assertEqual(df.iloc[0, 0], "A short name for the project.")
+        self.assertEqual(df.iloc[2, 0], "project.project_core.project_short_name")
 
         # note: it is difficult to extend this test to multiple tabs (too many expected column names to list here). Any ideas/thoughts about how to extend it welcome.
         expected_col_names = ["project.project_core.project_short_name", "project.project_core.project_title",
                               "project.project_core.project_description",
+                              "project.estimated_cell_count",
+                              "project.data_use_restrictions.text",
+                              "project.data_use_restrictions.ontology",
+                              "project.data_use_restrictions.ontology_label",
                               "project.supplementary_links", "project.insdc_project_accessions",
+                              "project.ega_accessions", "project.dbgap_accessions",
                               "project.geo_series_accessions", "project.array_express_accessions",
                               "project.insdc_study_accessions", "project.biostudies_accessions"]
         actual_col_names = list(df.iloc[2])
-        self.assertEqual(expected_col_names, actual_col_names)
\ No newline at end of file
+        self.assertEqual(expected_col_names, actual_col_names)
+
+    def tearDown(self) -> None:
+        if self.test_file:
+            if os.path.exists(self.test_file):
```

Comment:
I think you could use the tempfile library for tempfiles that automatically delete

Reply:
cool, very nice lib! thanks!